pkg/registry/apis/iam/team/store.go

```diff
@@ -119,6 +138,40 @@ func (s *LegacyStore) Get(ctx context.Context, name string, options *metav1.GetO
 	return &obj, nil
 }
 
+func (s *LegacyStore) Create(ctx context.Context, obj runtime.Object, createValidation rest.ValidateObjectFunc, options *metav1.CreateOptions) (runtime.Object, error) {
+	ns, err := request.NamespaceInfoFrom(ctx, true)
+	if err != nil {
+		return nil, err
+	}
+
+	teamObj, ok := obj.(*iamv0alpha1.Team)
+	if !ok {
+		return nil, fmt.Errorf("expected Team object, got %T", obj)
+	}
+
+	if createValidation != nil {
+		if err := createValidation(ctx, obj); err != nil {
+			return nil, err
+		}
+	}
+
+	createCmd := legacy.CreateTeamCommand{
```

Comment:
Could you please also add support for generating the name (`metadata.generateName`)? [Example](https://github.com/grafana/grafana/blob/main/pkg/registry/apis/iam/serviceaccount/store.go#L112-L115)